hbase-common/src/main/java/org/apache/hadoop/hbase/util/Threads.java

```diff
@@ -307,4 +308,33 @@ public static synchronized void printThreadInfo(PrintStream stream, String title
       throw new RuntimeException(e.getCause());
     }
   }
+
+  /**
+   * Checks whether any non-daemon thread is running.
+   * @return true if there are non daemon threads running, otherwise false
+   */
+  public static boolean isNonDaemonThreadRunning() {
+    AtomicInteger nonDaemonThreadCount = new AtomicInteger();
+    Set<Thread> threads =  Thread.getAllStackTraces().keySet();
+    for (Thread t: threads) {
```

Comment:
There is a small difference between master patch and branch-1 patch.
1. Replaced `threads.forEach(t -> {` with `for (Thread t: threads)`
2. Replaced `org.apache.commons.logging.Log` with `org.slf4j.LoggerFactory` since the former doesn't allow String formatter.

Cc @virajjasani